tensorflow_similarity/losses/arcface_loss.py

```diff
@@ -0,0 +1,232 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""ArcFace losses  base class.
+
+ArcFace: Additive Angular Margin Loss for Deep Face
+Recognition. [online] arXiv.org. Available at:
+<https://arxiv.org/abs/1801.07698v3>.
+
+"""
+
+from typing import Any, Callable, Dict, Optional, Tuple, Union
+
+import tensorflow as tf
+
+from tensorflow_similarity.utils import is_tensor_or_variable
+from tensorflow_similarity.algebra import build_masks
+from tensorflow_similarity.distances import Distance, distance_canonicalizer
+from tensorflow_similarity.types import FloatTensor, IntTensor
+
+from .metric_loss import MetricLoss
+from .utils import logsumexp
+from typing import Any, Callable, Union
+
+
+@tf.keras.utils.register_keras_serializable(package="Similarity")
+class ArcFaceLoss(MetricLoss):
+    """Implement of ArcFace: Additive Angular Margin Loss:
+            Step 1: Create a trainable kernel matrix with the shape of [embedding_size, num_classes].
+            Step 2: Normalize the kernel and prediction vectors.
+            Step 3: Calculate the cosine similarity between the normalized prediction vector and the kernel.
+            Step 4: Create a one-hot vector include the margin value for the ground truth class.
+            Step 5: Add margin_hot to the cosine similarity and multiply it by scale.
+            Step 6: Calculate the cross-entropy loss.
+
+            ArcFace: Additive Angular Margin Loss for Deep Face
+                     Recognition. [online] arXiv.org. Available at:
+                     <https://arxiv.org/abs/1801.07698v3>.
+
+            Standalone usage:
+                >>> loss_fn = tfsim.losses.ArcFaceLoss(num_classes=2, embedding_size=3)
+                >>> labels = tf.Variable([1, 0])
+                >>> embeddings = tf.Variable([[0.2, 0.3, 0.1], [0.4, 0.5, 0.5]])
+                >>> loss = loss_fn(labels, embeddings)
+            Args:
+                num_classes: Number of classes.
+                embedding_size: The size of the embedding vectors.
+                margin: The margin value.
+                scale: s in the paper, feature scale
+                name: Optional name for the operation.
+                reduction: Type of loss reduction to apply to the loss.
+            """
+
+    def __init__(self,
+                 num_classes: int,
+                 embedding_size: int,
+                 margin: float = 0.50,  # margin in radians
+                 scale: float = 64.0,  # feature scale
+                 name: Optional[str] = "ArcFaceLoss",
+                 reduction: Callable = tf.keras.losses.Reduction.AUTO,
+                 **kwargs: object) -> object:
+        super().__init__(reduction=reduction, name=name, **kwargs)

```

Comment:
I changed super().__init__(None, reduction=reduction, name=name, **kwargs) part and deleted the None. This needs to be solve the problem which occurs at testing pipeline